### PR TITLE
Tekton jib

### PR DESCRIPTION
**Pull request by iocanel**
This pull request changes the following:

### Major
Support for Jib image build when tool is maven/gradle and no Dockerfile present. In all other cases use kaniko.

### Minor
1. Aligns the parameter style with the one used in the tekton catalog (capital).
2. Git checkout directly in the root
3. Other minor improvements

### CODE REVIEW

File: annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddInsecureRegistryParamToTaskDecorator.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **/
+
+package io.dekorate.tekton.decorator;
+
+public class AddInsecureRegistryParamToTaskDecorator extends AddParamToTaskDecorator {
+
+  private static final String INSECURE_REGISTRY_PARAM_NAME = "INSECUREREGISTRY";
```
Comment (geoand) [inline annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddInsecureRegistryParamToTaskDecorator.java:22]:
```suggestion
  private static final String INSECURE_REGISTRY_PARAM_NAME = "INSECURE_REGISTRY";
```

File: annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddJibGradleBuildStepDecorator.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  private static final String BUILD_AND_PUSH = "build-and-push";
+  private static final String BUILDER_IMAGE_REF = "$(inputs.params.BUILDER_IMAGE)";
+  private static final String CMD = "gradle";
+  private static final String[] ARGS = {"jib", "--init-script=/tekton/home/init-script.gradle",  "-Djib.to.image=$(resources.outputs.image.url)", "-Djib.allowInsecureRegistries=$(params.INSECUREREGISTRY)", "-Djib.httpTimeout=60000"};
+
+  private static final String DOCKER_CONFIG = "DOCKER_CONFIG";
+  private static final String DOCKER_CONFIG_DEFAULT = "/tekton/home/.docker";
+
+  private static final String INSECUREREGISTRY = "INSECUREREGISTRY";
```
Comment (geoand) [inline annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddJibGradleBuildStepDecorator.java:38]:
```suggestion
  private static final String INSECUREREGISTRY = "INSECURE_REGISTRY";
```

File: annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddJibGradleBuildStepDecorator.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+import io.dekorate.DekorateException;
+import io.dekorate.deps.kubernetes.api.model.EnvVarBuilder;
+import io.dekorate.deps.tekton.pipeline.v1beta1.TaskSpecFluent;
+import io.dekorate.utils.Strings;
+
+public class AddJibGradleBuildStepDecorator extends NamedTaskDecorator implements StepDecorator {
+
+  private static final String BUILD_AND_PUSH = "build-and-push";
+  private static final String BUILDER_IMAGE_REF = "$(inputs.params.BUILDER_IMAGE)";
```
Comment (geoand) [inline annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddJibGradleBuildStepDecorator.java:31]: I assume this syntax if something Tekton specific?

File: annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddKaninkoBuildStepDecorator.java [hunk 4]
```diff
@@ -21,14 +21,13 @@
 import io.dekorate.deps.tekton.pipeline.v1beta1.TaskSpecFluent;
 import io.dekorate.kubernetes.decorator.Decorator;
 
-public class AddImageBuildStepDecorator extends NamedTaskDecorator implements StepDecorator{
-
+public class AddKaninkoBuildStepDecorator extends NamedTaskDecorator implements StepDecorator{
```
Comment (geoand) [inline annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddKaninkoBuildStepDecorator.java:24]:
```suggestion
public class AddKanikoBuildStepDecorator extends NamedTaskDecorator implements StepDecorator{
```

File: annotations/tekton-annotations/src/main/java/io/dekorate/tekton/handler/TektonHandler.java [hunk 5]
```diff
@@ -210,32 +212,35 @@ public void generateCommon(String group, TektonConfig config, ImageConfiguration
...
     resources.decorate(group, new AddParamToTaskDecorator(null, PATH_TO_CONTEXT_PARAM_NAME, PATH_TO_CONTEXT_DESCRIPTION, getContextPath(getProject())));
 
-    String monolithTaskName = monolithTaskName(config);
+    boolean isTask = group.equals(TEKTON_TASK);
 
-    String javaBuildTaskName = javaBuildTaskName(config);
-    String javaBuildStepName = javaBuildStepName(config);
-    String imageBuildTaskName = imageBuildTaskName(config);
-    String deployTaskName = deployTaskName(config);
+    final String monolithTaskName = monolithTaskName(config);
```
Comment (geoand) [inline annotations/tekton-annotations/src/main/java/io/dekorate/tekton/handler/TektonHandler.java:217]:
```suggestion
    final String monolithicTaskName = monolithTaskName(config);
```

File: annotations/tekton-annotations/src/main/resources/jib-gradle.sh [hunk 6]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+# Copied from tekton catalog!
+# Credits to: https://github.com/tektoncd/catalog/blob/master/jib-gradle/jib-gradle.yaml
+
+set -o errexit
+# Adds Gradle init script that applies the Jib Gradle plugin.
+echo "initscript {
```
Comment (geoand) [inline annotations/tekton-annotations/src/main/resources/jib-gradle.sh:8]: LOL

File: annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddInsecureRegistryParamToTaskDecorator.java (hunk 1 above)
Comment (iocanel) [reply annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddInsecureRegistryParamToTaskDecorator.java:22]: The idea is to use the same params with the equivalent task of the catalog, so I'd rather keep it as is.

File: annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddJibGradleBuildStepDecorator.java (hunk 2 above)
Comment (iocanel) [reply annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddJibGradleBuildStepDecorator.java:38]: Same

File: annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddJibGradleBuildStepDecorator.java (hunk 3 above)
Comment (iocanel) [reply annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddJibGradleBuildStepDecorator.java:31]: Yes

File: annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddInsecureRegistryParamToTaskDecorator.java (hunk 1 above)
Comment (geoand) [reply annotations/tekton-annotations/src/main/java/io/dekorate/tekton/decorator/AddInsecureRegistryParamToTaskDecorator.java:22]: OK

**Pull request closed without merging**